src/image/resamplingThread.js

```diff
@@ -396,6 +435,9 @@ export class ResamplingThread {
   ) {
     // We can't just pass in an Image or we would get a circular dependency
 
+    // Just needs to be resonably unique
+    const jobId = Date.now() + '' + Math.random();
```

Comment:
Could this be the volume id? That way it could be sent with the `imageresampled` event to have more context.

Reply:
It can't replace jobId, but it can be added along-side it. Updated to add the frame number (4D index) to the event, and handle only redrawing if it is the frame we are currently displaying.

Reply:
Looking good!